core/src/main/java/org/apache/calcite/plan/volcano/RuleQueue.java

```diff
@@ -494,11 +494,10 @@ VolcanoRuleMatch popMatch(VolcanoPlannerPhase phase) {
       }
     }
 
-    // A rule match's digest is composed of the operand RelNodes' digests,
-    // which may have changed if sets have merged since the rule match was
-    // enqueued.
-    match.recomputeDigest();
-
+    // If sets have merged since the rule match was enqueued, the match
+    // may not be removed from the matchMap because the subset may have
+    // changed, it is OK to leave it since the matchMap will be cleared
+    // at the end.
     phaseMatchList.matchMap.remove(
```

Comment:
Can we remove the relevant entry when subsets are merged/removed?
Otherwise, it looks like a memory leak.

Reply:
You have to remove all the matches for the subset in `matchMap`, find them in the `rulequeue` and remove them from `rulequeue`, and remove them from `names` set. I don't think it is worth sweating.
Since the leak already exists before this change, I prefer leaving it as it is.
I am going to propose a change to completely remove rulematch importance.

Reply:
The change does not fix the title, right ?

Reply:
What do you mean by `not fix the title`?